content/platforms/kubernetes/db-controller.md

```diff
@@ -0,0 +1,271 @@
+  and use databases via a database custom resource.
+weight: 35
+alwaysopen: false
+categories: ["Platforms"]
+aliases: /rs/concepts/kubernetes/k8s-operator-based-deployments
+---
+
+## Overview
+
+A database is created via a custom resource just like a cluster. The
+Redis Enterprise operator database controller will discover the custom resource,
+ensure its creation on the referenced cluster, and maintain parity between
+its specification and the database within the Redis Enterprise cluster.
+
+The custom resource defines the size and other facets of the desired database. For example,
+a 1GB database can simply be created on the `rec` cluster by the follow resource:
+
+```yaml
+kind: RedisEnterpriseDatabase
+metadata:
+  name: smalldb
+spec:
+  memory: 1GB
+  redisEnterpriseCluster:
+    name: rec
+```
+
+The cluster is referenced by name in the `redisEnterpriseCluster` and must exist
+in the same namespace.
+
+The database can then be created by simply apply the resource file (e.g., `smalldb.yaml`):
+
+```sh
+kubectl apply -f smalldb.yaml
+```
+
+The database controller will recognize the new custom resource and validate the
+specification. If the database specification is valid, the controller will
+create the database on the specified Redis Enterprise cluster. Once
+created, the database will be exposed via the same service mechanisms by
+the service rigger for the Redis Enterprise cluster.
+
+## Database lifecycle
+
+### Creating databases
+
+A database is created by creating a database custom resource in the same
+namespace as your cluster. For example:
+
+1. Create a file called db.yaml containing your database custom resource:
+
+    ```YAML
+    kind: RedisEnterpriseDatabase
+    metadata:
+      name: mydb
+    spec:
+      memory: 1GB
+      redisEnterpriseCluster:
+        name: rec
+    ```
+
+1. Apply the file to your namespace containing your cluster and the operator:
+
+    ```sh
+    kubectl apply -f db.yaml
+    ```
+
+1. Check for the status of your database:
+
+    ```sh
+    kubectl get redb/mydb -o yaml
+    ```
+
+    Once the status is `active`, the database is ready for use.
+
+{{% note %}}
+The `jq` tool is a useful tool for checking the status via the json output:
+
+```sh
+kubectl get redb/mydb -o json | jq .status.status
+```
+
+which will output the status value from all the JSON returned.
+
+{{% /note %}}
+
+
+### Modifying databases
+
+A database is modified by changing the custom resource for the database. You
+can simply edit your original specification and apply the change or use
+kubectl edit:
+
+1. Edit the definition:
+
+    ```sh
+    kubectl edit redb/mydb
+    ```
+
+2. Modify the specification (only properties in `spec` section) and save the changes.
+
+3. Monitor the status to see when the changes become effective:
+
+    ```sh
+    kubectl get redb/mydb -o yaml
+    ```
+
+    Once the status is `active`, the database is again ready for use.
+
+### Deleting databases
```

Comment:
We should include a section about the database finalizer, and how to delete REDB CRs if the REC was already deleted here.
This does not block merging this doc.
@alexmilowski let me know if you need more details about this finalizer

Reply:
Yes, please! Send me a description (or pointer to a doc) and I'll submit a separate PR with an update.